From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: diy-mtk.sh <build@localhost>
Date: Mon, 14 Jul 2026 00:00:00 +0000
Subject: [PATCH] net: ethernet: mtk_eth_soc: add mtkhnat Kconfig and Makefile
 (trimmed: Kconfig+Makefile only, no PPE guards, no soc hooks)

---
 drivers/net/ethernet/mediatek/Kconfig  | 24 ++++++++++++++++++++++++
 drivers/net/ethernet/mediatek/Makefile |  1 +
 2 files changed, 25 insertions(+)

--- a/drivers/net/ethernet/mediatek/Kconfig
+++ b/drivers/net/ethernet/mediatek/Kconfig
@@ -40,4 +40,28 @@ config NET_MEDIATEK_STAR_EMAC
 	  This driver supports the ethernet MAC IP first used on
 	  MediaTek MT85** SoCs.
 
+config MEDIATEK_NETSYS_V2
+	tristate "MediaTek Ethernet NETSYS V2 support"
+	depends on ARCH_MEDIATEK && NET_MEDIATEK_SOC
+	help
+	  This options enable MTK Ethernet NETSYS V2 support
+
+config MEDIATEK_NETSYS_V3
+	tristate "MediaTek Ethernet NETSYS V3 support"
+	depends on ARCH_MEDIATEK && NET_MEDIATEK_SOC
+	help
+	  This options enable MTK Ethernet NETSYS V3 support for
+	  XGMAC and USXGMII.
+
+config NET_MEDIATEK_HNAT
+	tristate "MediaTek HW NAT support"
+	depends on NET_MEDIATEK_SOC && NF_CONNTRACK && IP_NF_NAT
+	help
+	  This driver supports the Hardware Network Address Translation
+	  in the MediaTek MT7988/MT7986/MT7981 chipset family.
+
 endif #NET_VENDOR_MEDIATEK
--- a/drivers/net/ethernet/mediatek/Makefile
+++ b/drivers/net/ethernet/mediatek/Makefile
@@ -11,4 +11,5 @@ mtk_eth-$(CONFIG_NET_MEDIATEK_SOC_WED) += mtk_wed_debugfs.o
 endif
 obj-$(CONFIG_NET_MEDIATEK_SOC_WED) += mtk_wed_ops.o
 obj-$(CONFIG_NET_MEDIATEK_STAR_EMAC) += mtk_star_emac.o
+obj-$(CONFIG_NET_MEDIATEK_HNAT) += mtk_hnat/
 obj-$(CONFIG_NET_AIROHA) += airoha_eth.o
-- 
2.45.2
